### ISSUE TITLE
BBC iPlayer PlayList Unable to extract title/description

### ISSUE THREAD

**Issue by yepster** [rg3/youtube-dl#15710]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.02.26*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.02.26**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
youtube-dl -v --yes-playlist  https://www.bbc.co.uk/iplayer/episodes/b088f8wd
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'--yes-playlist', u'https://www.bbc.co.uk/iplayer/episodes/b088f8wd']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.02.26
[debug] Python version 2.7.3 (CPython) - Linux-3.2.0-4-amd64-x86_64-with-debian-7.11
[debug] exe versions: ffmpeg 3.1.1, ffprobe 3.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[bbc.co.uk:iplayer:playlist] b088f8wd: Downloading webpage
WARNING: unable to extract title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
WARNING: unable to extract description; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[download] Downloading playlist: b088f8wd
[bbc.co.uk:iplayer:playlist] playlist b088f8wd: Downloading 0 videos
[download] Finished downloading playlist: b088f8wd
...
<end of log>
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.bbc.co.uk/iplayer/episode/b0891b3x/rathad-na-deigheice-road-rescuers-episode-2
- Single video: https://www.bbc.co.uk/iplayer/episode/b08bz5yx/rathad-na-deigheice-road-rescuers-episode-3
- Playlist: https://www.bbc.co.uk/iplayer/episodes/b088f8wd

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---
Hallo,

The issue persists that after I upgraded to a certain version of youtube-dl  in Dec.2017,  youtube-dl was unable to extra title and description. I am running yt-dl on my UK server and single video can be successfully downloaded. 

I am not sure if it's iplayer website's upgrade in json format or something else from youtube-dl itself.

Thank you for the reply in advance

Best,
Ye

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/rg3/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.
If work on your *issue* requires account credentials please provide them or explain how one can obtain them.

**Comment by dirkf** [comment]
Still broken at https://github.com/ytdl-org/youtube-dl/commit/a40002444e64957594a1305bb2740fddb477beeb.

The page redirects to a BBC Alba playlist https://www.bbc.co.uk/programmes/b088f8wd, which doesn't have the same format as normal iPlayer playlists. As noted elsewhere, if a `programmes` page is a playlist, it needs  `/episodes` to be appended.

In the current version, the problem is at l.1390, where `playlist_data` is set to None:
```
        playlist_data = self._get_playlist_data(self._call_api(pid, 1))
```

**Comment by remitamine** [comment]
>The page redirects to a BBC Alba playlist https://www.bbc.co.uk/programmes/b088f8wd, which doesn't have the same format as normal iPlayer playlists. As noted elsewhere, if a programmes page is a playlist, it needs /episodes to be appended.
>
>In the current version, the problem is at l.1390, where playlist_data is set to None; the new code needs to allow for the case where _call_api() returns None.

the programme does not have any available episodes.

**Comment by dirkf** [comment]
True, but it does have episodes. The call to IBL (not unreasonably) returns None, which isn't handled, and should be.

Might the IBL call have failed even if the episodes had been available as clips and not on iPlayer?

**Comment by remitamine** [comment]
>True, but it does have episodes. The call to IBL (not unreasonably) returns None, which isn't handled, and should be.

it should report a better error message, but other then that i don't see much point in handling the redirection in what is eventually will fail anyway.

> Might the IBL call have failed even if the episodes had been available as clips and not on iPlayer?

i think when a user pass the iplayer episodes URL, it wouldn't be an expected behaviour to download the programme clips, if the user wants the clips, he can use the proper and explicit clips URL(https://www.bbc.co.uk/programmes/{pid}/clips).
